### PR TITLE
Doc: Fix typo in 2022-01-04-integer-shard-index.md

### PR DESCRIPTION
**Pull request by eltociear**
condtions -> conditions

### CODE REVIEW

**Review by ngaut — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Comment by Defined2014** [comment]
@eltociear Thanks for your commit. Please sign the cla

**Comment by eltociear** [comment]
@Defined2014 THX! I signed CLA.

File: docs/design/2022-01-04-integer-shard-index.md [hunk 1]
```diff
@@ -223,7 +223,7 @@ func (adder *exprPrefixAdder) addExprPrefix4DNFCond(condition *expression.Scalar
 
 - Function Process
   - Extract evrey binary expression from `condition` to a expression.Expression slice `dnfItems`.
-  - Make a new expression.Expression slice `newAccessItems` to store new condtions  after processing tidb_shard prefix.
+  - Make a new expression.Expression slice `newAccessItems` to store new conditions  after processing tidb_shard prefix.
```
Comment (Defined2014) [inline docs/design/2022-01-04-integer-shard-index.md:226]:
```suggestion
  - Make a new expression.Expression slice `newAccessItems` to store new conditions after processing tidb_shard prefix.
```

**Pull request merged**